### ISSUE TITLE
Using context manager fails if used more than once

### ISSUE THREAD

**Issue by ecederstrand** [RaRe-Technologies/sqlitedict#54]
```
>>> foo = SqliteDict('foo.sqlite')
>>> with foo:
...    foo[1] = 2
... 
>>> with foo:
...    foo[1] = 2
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/lib/python3.4/site-packages/sqlitedict.py", line 233, in __setitem__
    self.conn.execute(ADD_ITEM, (key, encode(value)))
AttributeError: 'NoneType' object has no attribute 'execute'
```

This is because `__exit__` calls `close()` which deletes `self.conn`.

it seems `__enter__` should check that `self.conn` exists and create it otherwise. I could just go through `__init__` once more, but `__init__` contains unnecessary actions like `CREATE TABLE ...`

I can create a patch if this issue is accepted.

**Comment by piskvorky** [comment]
Hmm, looks like a bug, thanks.

A PR would be welcome!

**Comment by ecederstrand** [comment]
Fixed in PR #55

**Comment by piskvorky** [comment]
Thanks @ecederstrand !
